### PR TITLE
Fixed advfort hatches (which were broken)

### PR DESCRIPTION
**Pull request by wolfboyft**
Also added support for pressure plates, maybe...?

### CODE REVIEW

**Comment by wolfboyft** [comment]
Oops, that should be "deselected trigger" not lever

**Comment by wolfboyft** [comment]
RE pressure plates, a fortress-made water-triggered pressure plate just opened a hatch (fortress-linked, so idk if linking in adventure works but it very likely does) in adventure mode when i dropped some water on it, so it seems it's just the stepping-on-them-as-an-adventurer that doesn't work.

And building them won't work either because we can only build triggerless plates with advfort. So I'll maybe make a contribution for that sometime :>

**Comment by warmist** [comment]
AFAIK stepping on pressure plates is intentionally disabled in adv mode. I think it's related to cage traps and stuff and/or your adventurer "knowing" the trap and intentionally not stepping on it.

**Comment by wolfboyft** [comment]
Oops. Didn't mean to do that.
Anyway I'll fix the error the test mentions and the "lever" thing.

File: gui/advfort.lua [hunk 1]
```diff
@@ -1109,16 +1109,17 @@ function LinkBuilding(args)
         end
     end
     if lever_bld==nil then
-        if bld:getType()==df.building_type.Trap and bld:getSubtype()==df.trap_type.Lever then
+        if bld:getType()==df.building_type.Trap and bld:getSubtype()==df.trap_type.Lever or bld:getSubtype()==df.trap_type.PressurePlate then
```
Comment (lethosor) [inline gui/advfort.lua:1112]: You should use parentheses here - `A and B or C` is equivalent to `(A and B) or C`, but you want `A and (B or C)`

Comment (wolfboyft) [reply gui/advfort.lua:1112]: Oops, you're right.

**Comment by wolfboyft** [comment]
Anything else need fixing?

**Comment by lethosor** [comment]
Nope, I just haven't had much time to try this out. Looks simple enough from a static review, though. Have you tested linking doors and hatches as well as other building types?

**Comment by wolfboyft** [comment]
Yee, & all seems good & cool!
Hatches and doors seem to be the only buildings that need a flag flipped to properly work with mechanisms (is bc they can work s/o)

**Review by lethosor — APPROVED** [review]

**Pull request merged**